### PR TITLE
Spec for global action IDs

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

⚠️ This spec has been moved from #6902. That version was branched off the new tab menu customization, and had a terribly convoluted git history. After discussion with the team, we've decided that it's best that this spec is merged atomically _first_, and used as the basis for #5888, as opposed to the other way around.

> This document is intended to serve as an addition to the [Command Palette Spec],
> as well as the [New Tab Menu Customization Spec].
> 
> As we come to rely more on actions being a mechanism by which the user defines
> "do something in the Terminal", we'll want to make it even easier for users to
> re-use the actions that they've already defined, as to reduce duplicated json as
> much as possible. This spec proposes a mechanism by which actions could be
> uniquely identifiable, so that the user could refer to bindings in other
> contexts without needing to replicate an entire json blob.
> 

## PR Checklist
* [x] Specs: #6899
* [x] References: #1571, #1912, #3337, #5025, #5524, #5633
* [x] I work here

## Detailed Description of the Pull Request / Additional comments
_\*<sup>\*</sup><sub>\*</sub> read the spec  <sub>\*</sub><sup>\*</sup>\*_

[Command Palette Spec]: https://github.com/microsoft/terminal/blob/master/doc/specs/%232046%20-%20Command%20Palette.md
[New Tab Menu Customization Spec]: https://github.com/microsoft/terminal/blob/master/doc/specs/%231571%20-%20New%20Tab%20Menu%20Customization.md

### CODE REVIEW

File: doc/specs/#6899 - Action IDs/#6899 - Action IDs.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+same `id`, it will replace the current value. In this way, users can redefine
+actions, or remove default ones (with something like `{ "id":
+"Terminal.OpenTab", "command":null }`
+
+We'd maintain a large list of default actions, each with unique `id`s set. These
+are all given `id`'s with a `Terminal.` prefix, to easily identify them as
+built-in, default actions. Not all of these actions will be given keys, but they
+will all be given `id`s.
+
+> 👉 NOTE: The IDs for the default actions will need to be manually created, not
```
Comment (leonMSFT) [inline doc/specs/#6899 - Action IDs/#6899 - Action IDs.md:82]: Just to clarify, when you say "manually created", do you mean we'll have to hard code it in the default settings?

File: doc/specs/#6899 - Action IDs/#6899 - Action IDs.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+last updated: 2020-07-22
+issue id: 6899
+---
+
+# Action IDs
+
+## Abstract
+
+This document is intended to serve as an addition to the [Command Palette Spec],
+as well as the [New Tab Menu Customization Spec].
```
Comment (DHowett) [inline doc/specs/#6899 - Action IDs/#6899 - Action IDs.md:13]: todo links?

File: doc/specs/#6899 - Action IDs/#6899 - Action IDs.md (hunk 1 above)
Comment (DHowett) [reply doc/specs/#6899 - Action IDs/#6899 - Action IDs.md:82]: yes

File: doc/specs/#6899 - Action IDs/#6899 - Action IDs.md (hunk 2 above)
Comment (zadjii-msft) [reply doc/specs/#6899 - Action IDs/#6899 - Action IDs.md:13]: These are actually footnote-style markdown links - see L218-219 to see where they're defined
image

They're defined at the bottom of the doc, so if I want to re-use the same text as a hyperlink elsewhere in the doc, I don't have to reprint the URL, or even have the URL inline. 

Handy for specs like this IMO

**Pull request merged**